src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs

```diff
@@ -0,0 +1,589 @@
+                        }
+                    }
+
+                    if (skippedTestModes == Xunit.RuntimeTestModes.Any)
+                    {
+                        testInfos = FilterForSkippedRuntime(testInfos, (int)Xunit.TestRuntimes.CoreCLR, options);
+                    }
+                    testInfos = DecorateWithSkipOnPlatform(testInfos, (int)skippedTestPlatforms, options);
+                    testInfos = DecorateWithSkipOnCoreClrConfiguration(testInfos, skippedTestModes, skippedConfigurations);
+
+                    break;
+            }
+        }
+
+        return testInfos;
+    }
+
+    private static ImmutableArray<ITestInfo> DecorateWithSkipOnCoreClrConfiguration(ImmutableArray<ITestInfo> testInfos, Xunit.RuntimeTestModes skippedTestModes, Xunit.RuntimeConfiguration skippedConfigurations)
+    {
+        const string ConditionClass = "TestLibrary.CoreClrConfigurationDetection";
+        List<string> conditions = new();
+        if (skippedConfigurations.HasFlag(Xunit.RuntimeConfiguration.Debug | Xunit.RuntimeConfiguration.Checked | Xunit.RuntimeConfiguration.Release))
+        {
+            // If all configurations are skipped, just skip the test as a whole
+            return ImmutableArray<ITestInfo>.Empty;
+        }
+
+        if (skippedConfigurations.HasFlag(Xunit.RuntimeConfiguration.Debug))
+        {
+            conditions.Add($"!{ConditionClass}.IsDebugRuntime");
+        }
+        if (skippedConfigurations.HasFlag(Xunit.RuntimeConfiguration.Checked))
+        {
+            conditions.Add($"!{ConditionClass}.IsCheckedRuntime");
+        }
+        if (skippedConfigurations.HasFlag(Xunit.RuntimeConfiguration.Release))
+        {
+            conditions.Add($"!{ConditionClass}.IsReleaseRuntime");
+        }
+        if (skippedTestModes.HasFlag(Xunit.RuntimeTestModes.RegularRun))
+        {
+            conditions.Add($"{ConditionClass}.IsStressTest");
+        }
+        if (skippedTestModes.HasFlag(Xunit.RuntimeTestModes.JitStress))
+        {
+            conditions.Add($"!{ConditionClass}.IsJitStress");
+        }
+        if (skippedTestModes.HasFlag(Xunit.RuntimeTestModes.JitStressRegs))
+        {
+            conditions.Add($"!{ConditionClass}.IsJitStressRegs");
+        }
+        if (skippedTestModes.HasFlag(Xunit.RuntimeTestModes.JitMinOpts))
+        {
+            conditions.Add($"!{ConditionClass}.IsJitMinOpts");
+        }
+        if (skippedTestModes.HasFlag(Xunit.RuntimeTestModes.TailcallStress))
+        {
+            conditions.Add($"!{ConditionClass}.IsTailcallStress");
+        }
+        if (skippedTestModes.HasFlag(Xunit.RuntimeTestModes.ZapDisable))
+        {
+            conditions.Add($"!{ConditionClass}.IsZapDisable");
+        }
+
+        if (skippedTestModes.HasFlag(Xunit.RuntimeTestModes.AnyGCStress))
+        {
+            conditions.Add($"!{ConditionClass}.IsGCStress");
+        }
+        else if (skippedTestModes.HasFlag(Xunit.RuntimeTestModes.GCStress3))
```

Comment:
Hmm, is the distinction between the individual if statements and the if-else if-else if structure for the GC stress modes required? I'm worried that this asks for maintenance trouble, when someone wants to add extra skip modes in the future, they'll have trouble figuring out which method to use. If the distinction is necessary, it might be at least worth a code comment.

Reply:
The problem is that the `GCStress` flag is equivalent to `GCStress3 | GCStressC`, so we need to handle it separately to get accurate results when someone is using another GCStress mode. Also, if we just continue with the `if` chain, then every check for GCStress would be `IsGCStress || IsGCStress3 || IsGCStressC` which is extra IL that we don't need to emit. I'll add a comment.